packages/kbn-failed-test-reporter-cli/failed_tests_reporter/generate_scout_test_failure_artifacts.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import Path from 'path';
+
+import globby from 'globby';
+import fs from 'fs';
+import { createHash } from 'crypto';
+import { ToolingLog } from '@kbn/tooling-log';
+import { BuildkiteMetadata } from './buildkite_metadata';
+
+const SCOUT_TEST_FAILURE_DIR_PATTERN = '.scout/reports/scout-playwright-test-failures-*';
+const SUMMARY_REPORT_FILENAME = 'test-failures-summary.json';
+
+export async function generateScoutTestFailureArtifacts({
```

Comment:
This function suppose to update html reports with github issue link and also create new GH issue, if needed. I think PR is getting too large and I think can be done in the follow-up